pkg/auto/lights/auto.go

```diff
@@ -181,25 +185,32 @@ func (b *BrightnessAutomation) processStateChanges(ctx context.Context, readStat
 			}
 
 			// if the context remains live, schedule another update soon
-			// TODO: make this delay configurable, or add backoff etc.
-			after := 5 * time.Second
+			b.retries++
+			after := time.Duration(b.retries) * b.backoffMultiplier
 			b.logger.Error("processState failed; scheduling retry",
 				zap.Error(err),
 				zap.Duration("retryAfter", after),
 			)
 			ttl = after
 		}
 
-		if ttl < 0 {
-			b.logger.Warn("ttl < 0; using refreshEvery instead")
+		if err == nil || b.retries > b.maxRetries {
+			b.retries = 0
 		}
```

Comment:
By the name, I would expect that `maxRetries` is intended to limit the number of time a retry happens. 
But when `retries` exceeds `maxRetries`, all that happens is `retries` is set back to 0, which will reset the backoff duration calculated above but the retry will still happen.